packages/turbo-codemod/__tests__/add-package-names.test.ts

```diff
@@ -0,0 +1,117 @@
+import { setupTestFixtures } from "@turbo/test-utils";
+import { transformer } from "../src/transforms/add-package-names";
+
+describe("add-package-names", () => {
+  const { useFixture } = setupTestFixtures({
+    directory: __dirname,
+    test: "add-package-names",
+  });
+
+  test("missing names", async () => {
+    // load the fixture for the test
+    const { root, readJson } = useFixture({
+      fixture: "missing-names",
+    });
+
+    // run the transformer
+    const result = await transformer({
+      root,
+      options: { force: false, dry: false, print: false },
+    });
+
+    // result should be correct
+    expect(result.fatalError).toBeUndefined();
+    expect(result.changes).toMatchInlineSnapshot(`
+      Object {
+        "packages/ui/package.json": Object {
+          "action": "modified",
+          "additions": 1,
+          "deletions": 0,
+        },
+        "packages/utils/package.json": Object {
+          "action": "modified",
+          "additions": 1,
+          "deletions": 0,
+        },
+      }
+    `);
+
+    // validate unique names
+    const names = new Set();
+
+    for (const pkg of ["ui", "utils"]) {
+      const pkgJson = readJson<{ name: string }>(
+        `packages/${pkg}/package.json`
+      );
+      expect(pkgJson?.name).toBeDefined();
+      expect(names.has(pkgJson?.name)).toBe(false);
+      names.add(pkgJson?.name);
+    }
+  });
+
+  test("duplicate names", async () => {
```

Comment:
FWIW we've never supported duplicate package names in the workspace

Reply:
Yea - this was an easy add since if we're generating new names we need to ensure they're unique, and if we're ensuring new ones are unique we should ensure they're all unique. 

There still might be an edge case or two here, but this should be good for most cases.